chrome/views/popupView.js

```diff
@@ -63,68 +66,128 @@ GmailToTrello.PopupView.prototype.confirmPopup = function() {
+        else {
+            buttonText = minDue.toString('MMM d, yyyy');
         }
-        gtt_log('PopupView:confirmPopup: adding Button and Popup');
-        $button.first().appendTo(this.$toolBar);
-        $popup.first().appendTo(this.$toolBar);
     }
 
-    if (needInit || $popup.length < 1) {
-        if (this.html && this.html['popup'] && this.html['popup'].length > 0) {
-            gtt_log('PopupView:confirmPopup: adding popup');
-            this.$toolBar.append(this.html['popup']);
-            needInit = true;
-        } else {
-            needInit = false;
-            $.get(chrome.extension.getURL('views/popupView.html'), function(data) {
-                // data = self.parent.replacer(data, {'jquery-ui-css': chrome.extension.getURL('lib/jquery-ui-1.12.1.min.css')}); // OBSOLETE (Ace@2017.06.09): Already loaded by manifest
-                self.html['popup'] = data;
-                gtt_log('PopupView:confirmPopup: creating popup')
-                self.$toolBar.append(data);
-                self.parent.loadSettings(self); // Calls init_popup
-            });
-        }
+    var img = 'GtT';
+    // Refresh icon present? If so, use graphics, if not, use text:
+    if ($('div.asl.T-I-J3.J-J5-Ji,div.asf.T-I-J3.J-J5-Ji', self.$toolBar).length > 0) {
+        img = '<img class="f tk3N6e-I-J3" src="'
+          + chrome.extension.getURL('images/trello-icon-black.png')
+          + '" />';
     }
 
-    if (needInit) {
-        this.parent.loadSettings(this); // Calls init_popup
+    self.html['add_to_trello'] =
+        '<div id="gttButton" class="T-I J-J5-Ji ar7 nf T-I-ax7 L3' + className + '" ' // "lS T-I-ax7 ar7"
+          + 'data-tooltip="Add this Gmail to Trello">'
+          + '<div aria-haspopup="true" role="button" class="J-J5-Ji W6eDmd L3 J-J5-Ji Bq L3" tabindex="0">' // class="J-J5-Ji W6eDmd L3 J-J5-Ji Bq L3">' // 
+          + img
+          + '<span id="gttButtonText"></span>'
+          + (buttonText.length > 0 ? '&nbsp;' + buttonText + '&nbsp;' : '')
+          + '<div id="gttDownArrow" class="G-asx T-I-J3 J-J5-Ji">&#9662;</div></div></div>';
+
+    self.confirmPopup();
+
+};
+
+GmailToTrello.PopupView.prototype.addOrCreatePopup = function () {
+    var self = this;
+    if (self.html && self.html['popup'] && self.html['popup'].length > 0) {
+        gtt_log('PopupView:addOrCreatePopup: adding popup');
+        self.$toolBar.append(self.html['popup']);
+        this.parent.loadSettings(self); // Calls init_popup
+    } else {
+        $.get(chrome.extension.getURL('views/popupView.html'), function(data) {
+            // data = self.parent.replacer(data, {'jquery-ui-css': chrome.extension.getURL('lib/jquery-ui-1.12.1.min.css')}); // OBSOLETE (Ace@2017.06.09): Already loaded by manifest
+            self.html['popup'] = data;
+            gtt_log('PopupView:addOrCreatePopup: creating popup')
+            self.$toolBar.append(data);
+            self.parent.loadSettings(self); // Calls init_popup
+        });
```

Comment:
This is the same as it was before, just pulled out into a separate function since it happens at a different time in the sequence.